src/test_rig_bluesky/plans.py

```diff
@@ -10,25 +10,43 @@
 from dodal.devices.motors import XYZStage
 from dodal.plan_stubs.data_session import attach_data_session_metadata_decorator
 from dodal.plans import spec_scan
-from ophyd_async.core import TriggerInfo
+from ophyd_async.core import Device, TriggerInfo, YamlSettingsProvider
 from ophyd_async.epics.adaravis import AravisDetector
 from ophyd_async.epics.adcore import NDAttributePv, NDAttributePvDbrType
 from ophyd_async.epics.adcore._core_io import NDROIStatNIO
-from ophyd_async.plan_stubs import setup_ndattributes
+from ophyd_async.plan_stubs import (
+    apply_settings,
+    apply_settings_if_different,
+    retrieve_settings,
+    setup_ndattributes,
+    store_settings,
+)
 from scanspec.specs import Line, Spec
 
 imaging_detector = inject("imaging_detector")
 spectroscopy_detector = inject("spectroscopy_detector")
 sample_stage = inject("sample_stage")
 
+yaml_directory = os.path.abspath("./src/test_rig_bluesky/")
 
-@dataclass
-class ROI:
-    channel: int
-    name: str
-    start_x: int
-    start_y: int
-    size: int
+
+def save_settings(
+    device: Device,
+    yaml_filename: str,
+    yaml_directory: str = yaml_directory,
+):
+    provider = YamlSettingsProvider(yaml_directory)
+    yield from store_settings(provider, yaml_filename, device)
+
+
+def load_settings(
+    device: Device,
+    yaml_filename: str,
+    yaml_directory: str = yaml_directory,
+):
+    provider = YamlSettingsProvider(yaml_directory)
+    settings = yield from retrieve_settings(provider, yaml_filename, device)
+    yield from apply_settings_if_different(settings, apply_settings)
```

Comment:
The design of ophyd-async is intended to be storage agnostic. The idea is that it may be using YAML files now but in future we may switch to a different storage medium (e.g. a database), that's why there's a separate provider object and name parameter, and also why the name parameter excludes the `.yaml` suffix. So you don't want to tie the parameters to YAML from the user's point of view. I would suggest hard-coding the directory and changing the parameter_name, something like this:


```suggestion
def save_settings(
    device: Device,
    design_name: str,
):
    provider = YamlSettingsProvider(yaml_directory)
    yield from store_settings(provider, design_name, device)


def load_settings(
    device: Device,
    design_name: str,
):
    provider = YamlSettingsProvider(yaml_directory)
    settings = yield from retrieve_settings(provider, design_name, device)
    yield from apply_settings_if_different(settings, apply_settings)
```

If you do it this way you can later switch out from YAML to something else and the end user would see no change, they would still be inputting the same parameters.